url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/ArgCommand.java

```diff
@@ -0,0 +1,244 @@
+	public static final int COMMAND7 = 7;
+	public static final int COMMAND8 = 8;
+	public static final int MAXINDEX = 8;
+	private static Object[] clis = new Object[MAXINDEX + 1];
+	private static ArgCommand[] argCommands = new ArgCommand[MAXINDEX + 1];
+	private static int previousLevel = 0;
+	private static String programOptionSuffix;
+	private int currentLevel = 0;
+   
+    /**
+     * Protected no argument constructor.
+     */
+    protected ArgCommand () {
+    }
+    
+    /**
+     * Creates a new {@code ArgCommand} instance.
+     * 
+     * @return the new {@code ArgCommand} instance
+     * @see #create(String)
+     * @see #create(String[])
+     */
+    public static ArgCommand create () {
+    	ArgCommand argCommand = new ArgCommand();
+    	argCommand.currentLevel = previousLevel++;
+    	argCommand.setArgCommand(argCommand);
+        return argCommand;
+    }
+    
+    /**
+     * Creates a new {@code ArgCommand} instance and sets the
+     * usage text specification.
+     * 
+     * @param usageText the usage text as a String
+     * @return the new {@code ArgCommand} instance
+     * @see #create()
+     * @see #create(String[])
+     * @see #setUsageText(String)
+     */
+    public static ArgCommand create (String usageText) {
+        ArgCommand argCommand = create();
+        argCommand.setUsageText(usageText);
+        return argCommand;
+    }
+    
+    /**
+     * Creates a new {@code ArgCommand} instance and sets the
+     * usage text specification.
+     * 
+     * @param usageText the usage text as an array of Strings
+     * @return the new {@code ArgCommand} instance
+     * @see #create()
+     * @see #create(String)
+     * @see #setUsageText(String[])
+     */
+    public static ArgCommand create (String[] usageText) {
+    	ArgCommand argCommand = create();
+        argCommand.setUsageText(usageText);
+        return argCommand;
+    }
+    
+    public static ArgCommand create (String usageText, Object cli) {
+        ArgCommand argCommand = create(usageText);
+        argCommand.setCLI(cli);
+        if (argCommand.parseUsage(cli)) {
+        	return argCommand;
+        }
+        return null;
+    }
+    
+    public static ArgCommand create (String [] usageText, Object cli) {
+        ArgCommand argCommand = create(usageText);
+        argCommand.setCLI(cli);
+        if (argCommand.parseUsage(cli)) {
+            return argCommand;
+        }
+        return null;
+    }
+    
+    public int parse (String [] args) {
+        int nArg = parseArguments(args);
+        if (nArg < 0) {
+        	return nArg;
+        }
+        return nArg;
+    }
+    
+    public boolean executeCommand () {
+    	String command = value("<command>");
+    	if (command != null) {
+    		String [] commandArgs = valueArray("<command-args>");
+    		return executeCommand(command, commandArgs);
+    	}
+    	return false;
+    }
+   
+	public boolean executeCommand(String command, String [] commandArgs) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 153-173 into a separate method.